packages/node/src/app/analytics-node.ts

```diff
@@ -0,0 +1,345 @@
+import {
+  EventProperties,
+  Traits,
+  Emitter,
+  CoreAnalytics,
+  CoreContext,
+  CorePlugin,
+  EventFactory,
+  EventQueue,
+  dispatch,
+  resolvePageArguments,
+  PageParams,
+  Integrations,
+  CoreOptions,
+  Callback,
+  CoreSegmentEvent,
+  bindAll,
+} from '@segment/analytics-core'
+
+import { version } from '../../package.json'
+
+/** create a derived class since we may want to add node specific things to Context later  */
+export class ContextNode extends CoreContext {}
+
+export type SegmentEventOptionsNode = CoreOptions & Identity
+
+export type Identity =
+  | { userId: string; anonymousId?: string }
+  | { userId?: string; anonymousId: string }
+
+type SegmentEventTypeNode = 'track' | 'page' | 'identify' | 'alias' | 'screen'
+
+export interface SegmentEventNode extends CoreSegmentEvent {
```

Comment:
Minor: `NodeSegmentEvent` feels like a more natural extension of `CoreSegmentEvent` to me since then we're always just modifying the prefix.

Reply:
I've been doing Noun_Suffix (e.g. SegmentEventTypeNode, etc) because AnalyticsNode uses "Noun_Suffix" /shrug

Reply:
Yeah, you can take or leave my suggestion :smile: I just wondered why it wasn't SegmentEventCore then :D

Reply:
I think it was a good one! Updated.